ARCHITECTURE.md

```diff
@@ -0,0 +1,231 @@
+imagePullPolicy| [ImagePullPolicy](https://kubernetes.io/docs/concepts/containers/images/)
+priorityClassName | [priority class name](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/#priorityclass)
+nodeSelector| [node selector](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector)
+affinity | [affinity and anti-affinity](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity)
+
+These K8s setting are available for each component under the `k8s` field, for example:
+
+```yaml
+trafficManagement:
+  components:
+    pilot:
+      k8s:
+        hpaSpec:
+          # HPA spec, as defined in K8s API
+```
+
+### Translations
+
+API translations are version specific and are expressed as a
+[table of Translators](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/translate/translate.go#L89)
+indexed by minor [version](https://github.com/istio/operator/blob/master/pkg/version/version.go). This is because
+mapping rules are only allowed to change between minor (not patch) versions.
+
+The new API fields are translated to the output manifest in two ways:
+
+1. The new API fields are mapped to the old Helm values.yaml schema using the
+[APIMapping](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/translate/translate.go#L91)
+field of the [Translator](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/translate/translate.go#L48)
+struct.
+1. The K8s settings are applied to resources in the output manifest using the
+[KubernetesMapping](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/translate/translate.go#L108)
+field in the [Translator](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/translate/translate.go#L48)
+struct.
+
+Other per-component mappings to Helm values.yaml are expressed in the
+[ComponentMaps](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/translate/translate.go#L64)
+struct.
+
+### Validations
+
+Both the new API and old Helm API are validated. The new API is validated through a table of validation rules in
+[pkg/validate/validate.go](https://github.com/istio/operator/blob/master/pkg/validate/validate.go). These rules
+refer to the Go struct path schema and hence have names with a capitalized first letter.
+The old Helm values.yaml API is validated in
+[validate_values.go](https://github.com/istio/operator/blob/master/pkg/validate/validate_values.go)
+and refer to the values.yaml data paths. Hence, these rules have names with a lower case first letter.
+
+## K8s controller
+
+TODO(rcernich).
+
+## Manifest creation
+
+Manifest rendering is a multi-step process, shown in the figure below. ![rendering
+process](images/operator_render_flow.svg) The example in the figure shows the rendering being triggered by a CLI `iop`
```

Comment:
The diagram mentions "remote release URL," so I assume that the operator has the ability to download Helm charts? How does it work? Where are the charts stored? Are they periodically refreshed? How? Is the operator's filesystem read-only?

If you don't intend to document this feature, remove it from the diagram.

Reply:
The feature is still not fully baked but will be there in the release. I've put a TODO here for now.